pageserve/app.py

```diff
@@ -29,14 +29,17 @@ def index():
     is_auth = has_edit_access(get_user_info(user,
                                             app.config['USERS_ENDPOINT']
                                             + "authorization"))
-    posts = get_posts()
-    return render_template(
-        'index.html',
-        posts=posts,
-        auth=is_auth,
-        user=user,
-        app_config=app.config
-    )
+    try:
+        posts = get_posts()
+        return render_template(
+            'index.html',
+            posts=posts,
+            auth=is_auth,
+            user=user,
+            app_config=app.config
+        )
+    except RuntimeError:
+        return "Unable to get posts from posts service.", 500
```

Comment:
Since you are giving a message to your RuntimeError on line 117, maybe it would be better to just pass the error along instead of redefining a new error message. This way, both `except`s in `index()` and `show_events()` would look the same.
```python
except RuntimeError as error:
    return str(error), 500
```